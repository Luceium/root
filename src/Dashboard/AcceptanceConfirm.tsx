import React from "react";
import { HACKATHON_YEAR } from "../constants";
export default () => <div>
    <h4>Congratulations! You have been selected to attend Treehacks {HACKATHON_YEAR}!</h4>
    <p>You have until February 1st to confirm your attendance at Treehacks {HACKATHON_YEAR}. If you don't confirm by then, we'll assume you can't make it and give your spot to someone else.</p>
    <button className="btn btn-custom inverted">decline spot</button>
    <button className="btn btn-custom">confirm spot</button>
    <p>We are committed to helping every admitted hacker get here! Check out the travel section for details.</p>
</div>;