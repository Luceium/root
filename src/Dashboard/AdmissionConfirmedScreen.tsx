import React from "react";
import { HACKATHON_YEAR } from "../constants";

export default () => <div>
    <p>Admission confirmed! See you there.</p>
</div>;