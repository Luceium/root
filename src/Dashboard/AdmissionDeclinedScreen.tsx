import React from "react";
import { HACKATHON_YEAR } from "../constants";

export default () => <div className="admitted-content">
    <p>You were selected for TreeHacks {HACKATHON_YEAR} but declined admission. We hate to see you go, but we appreciate your response and will be giving your spot to someone else.</p>
</div>;