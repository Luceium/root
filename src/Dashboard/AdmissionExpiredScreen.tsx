import React from "react";
import { HACKATHON_YEAR } from "../constants";

export default () => <div className="admitted-content">
    <p>You were selected to attend TreeHacks {HACKATHON_YEAR} but you did not confirm your attendance before the deadline, so we have given your spot to another hacker. Hope to see you next year!</p>
</div>;