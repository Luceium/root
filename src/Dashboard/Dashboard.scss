.dashboard {
    width: 100%;
    height: 100vh;
    background-repeat: no-repeat;

	.dashboard-design {
		color: white;
        text-align: center;
        background-color: #686e77;
        width: 100%;
        max-width: 500px;
        padding: 20px;
	}
}

/*.dashboard-body {
    background-image: url(/art/circuit_left.svg), url(/art/circuit_bottom.svg);
    background-position: left top, right bottom;
    background-repeat: no-repeat, no-repeat;
    height: 100%;
}*/