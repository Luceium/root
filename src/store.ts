import { Reducer, createStore, applyMiddleware } from 'redux';
import { reducers } from './store/index';
import thunkMiddleware from 'redux-thunk'
import { connectRouter, routerMiddleware, LOCATION_CHANGE } from 'connected-react-router';
import GoogleAnalyticsGtag, { trackPageView, trackEvent } from '@redux-beacon/google-analytics-gtag';
import { createMiddleware } from "redux-beacon";
import history from "./history";

declare const GA_TRACKING_ID: string;
const ga = GoogleAnalyticsGtag(GA_TRACKING_ID);
const eventsMap = {
  [LOCATION_CHANGE]: trackPageView(action => ({
    path: action.payload.routerState.url
  })),
  '*': trackEvent(action => ({
    category: 'redux',
    action: action.type,
  })),
}
const gaMiddleware = createMiddleware(eventsMap, ga);

const store = createStore(
  connectRouter(history)(reducers),
  applyMiddleware(thunkMiddleware, routerMiddleware(history), gaMiddleware)
);
export default store;