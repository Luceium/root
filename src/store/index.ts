import { combineReducers, Reducer } from 'redux';

import authReducer from './auth/reducer';
import baseReducer from './base/reducer';

export const reducers: Reducer = combineReducers({
  auth: authReducer,
  base: baseReducer
});
