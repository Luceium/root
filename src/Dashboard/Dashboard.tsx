import React from "react";
import { connect } from "react-redux";
import { getUserProfile } from "../store/form/actions";
import { IFormState } from "../store/form/types";
import { IDashboardProps, IDashboardWrapperProps } from "./types";
import Loading from "../Loading/Loading";
import "./Dashboard.scss";

function getDeadline(type) {
    switch (type) {
        case "is":
            return new Date("2018-11-27T07:59:00.000Z");
        case "stanford":
            return new Date("2019-02-14T07:59:00.000Z");
        case "oos":
        default:
            return new Date("2018-11-20T07:59:00.000Z");
    }
  }

function formatDate(date) {
    var monthNames = [
        "January", "February", "March",
        "April", "May", "June", "July",
        "August", "September", "October",
        "November", "December"
    ];

    var day = date.getDate();
    var monthIndex = date.getMonth();
    var year = date.getFullYear();

    return monthNames[monthIndex] + ' ' + day + ', ' + year;
}


export const Dashboard = (props: IDashboardProps) => {
    let date = getDeadline(props.profile.type);
    const dateNow = new Date();
    const diffDays = Math.round(Math.abs((date.getTime() - dateNow.getTime()) / (24 * 60 * 60 * 1000)));
    const deadline = formatDate(date);
    return (
        <div className="dashboard" style={{ "backgroundImage": `url('${require('../art/combined_circuit.svg')}')` }}>
            <div style={{ position: 'absolute', top: "50%", left: "50%", transform: "translateX(-50%) translateY(-50%)" }}>
                <div className="dashboard-design">
                    {
                        props.profile.status === "submitted" ? (
                            <span>
                                Your application has been received &ndash; you are all good for now!<br /><br />We will email you when decisions are released and will handle any travel questions at that time. Thanks for applying :)</span>
                        ) : (
                                <div>
                                    <div>
                                        You haven't submitted your application yet. You have
                            </div>
                                    <div style={{ color: "#00E073", fontSize: '70px', marginBottom: -15 }}>
                                        {diffDays}
                                    </div>
                                    <div>
                                        days to submit your application before the deadline:<br /><strong>{deadline}</strong>.
                            </div>
                                </div>
                            )
                    }
                </div>
            </div>
        </div>
    );
}

const mapStateToProps = state => ({
    ...state.form
});

const mapDispatchToProps = (dispatch, ownProps) => ({
    getUserProfile: () => dispatch(getUserProfile())
});

class DashboardWrapper extends React.Component<IDashboardWrapperProps, {}> {
    componentDidMount() {
        this.props.getUserProfile();
    }
    render() {
        if (!this.props.profile) {
            return <Loading />;
        }
        return <Dashboard profile={this.props.profile} />;
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(DashboardWrapper);
