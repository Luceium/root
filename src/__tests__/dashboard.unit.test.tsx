import React from "react";
import { shallow, mount, render } from 'enzyme';
import { Provider } from "react-redux";
import store from "../store";
import sinon from "sinon";
import {Dashboard} from "../Dashboard/Dashboard";

const literallyJustDateNow = () => Date.now();

// test('It should call and return Date.now()', () => {
//   const realDateNow = Date.now.bind(global.Date);
//   const dateNowStub = jest.fn(() => 1530518207007);
//   global.Date.now = dateNowStub;

//   expect(literallyJustDateNow()).toBe(1530518207007);
//   expect(dateNowStub).toHaveBeenCalled();

//   global.Date.now = realDateNow;
// });

it('renders dashboard', () => {
    //   store.dispatch(action())
    //   const wrapper = render(
    //     <Provider store={store}><PaymentHistory /></Provider>
    //   );
    const profile = {

    };
    const wrapper = render(
        <Dashboard profile={ profile } />
    );
    // expect(wrapper).toMatchSnapshot();
    //   expect(wrapper.text()).toContain("No rows found");
});