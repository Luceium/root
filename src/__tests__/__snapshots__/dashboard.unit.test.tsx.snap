// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dashboard admission confirm expired 1`] = `
<div
  className="dashboard"
  style={
    Object {
      "backgroundImage": "url('function SvgCombined_circuit(props) {
  return React.createElement(
    'svg', 
    Object.assign({}, props, {'data-file-name': SvgCombined_circuit.name})
  );
}')",
    }
  }
>
  <div
    style={
      Object {
        "left": "50%",
        "position": "absolute",
        "top": "50%",
        "transform": "translateX(-50%) translateY(-50%)",
      }
    }
  >
    <div
      className="dashboard-design"
    >
      <Component />
    </div>
  </div>
</div>
`;

exports[`dashboard admission confirmed 1`] = `
<div
  className="dashboard"
  style={
    Object {
      "backgroundImage": "url('function SvgCombined_circuit(props) {
  return React.createElement(
    'svg', 
    Object.assign({}, props, {'data-file-name': SvgCombined_circuit.name})
  );
}')",
    }
  }
>
  <div
    style={
      Object {
        "left": "50%",
        "position": "absolute",
        "top": "50%",
        "transform": "translateX(-50%) translateY(-50%)",
      }
    }
  >
    <div
      className="dashboard-design"
    >
      <Connect(Component)
        confirmedYet={true}
      />
    </div>
  </div>
</div>
`;

exports[`dashboard admission declined 1`] = `
<div
  className="dashboard"
  style={
    Object {
      "backgroundImage": "url('function SvgCombined_circuit(props) {
  return React.createElement(
    'svg', 
    Object.assign({}, props, {'data-file-name': SvgCombined_circuit.name})
  );
}')",
    }
  }
>
  <div
    style={
      Object {
        "left": "50%",
        "position": "absolute",
        "top": "50%",
        "transform": "translateX(-50%) translateY(-50%)",
      }
    }
  >
    <div
      className="dashboard-design"
    >
      <Component />
    </div>
  </div>
</div>
`;

exports[`dashboard admitted 1`] = `
<div
  className="dashboard"
  style={
    Object {
      "backgroundImage": "url('function SvgCombined_circuit(props) {
  return React.createElement(
    'svg', 
    Object.assign({}, props, {'data-file-name': SvgCombined_circuit.name})
  );
}')",
    }
  }
>
  <div
    style={
      Object {
        "left": "50%",
        "position": "absolute",
        "top": "50%",
        "transform": "translateX(-50%) translateY(-50%)",
      }
    }
  >
    <div
      className="dashboard-design"
    >
      <Connect(Component)
        confirmedYet={false}
        deadline="2048-01-30T04:39:47.512Z"
      />
    </div>
  </div>
</div>
`;

exports[`dashboard incomplete after deadline 1`] = `
<div
  className="dashboard"
  style={
    Object {
      "backgroundImage": "url('function SvgCombined_circuit(props) {
  return React.createElement(
    'svg', 
    Object.assign({}, props, {'data-file-name': SvgCombined_circuit.name})
  );
}')",
    }
  }
>
  <div
    style={
      Object {
        "left": "50%",
        "position": "absolute",
        "top": "50%",
        "transform": "translateX(-50%) translateY(-50%)",
      }
    }
  >
    <div
      className="dashboard-design"
    >
      <span>
        Sorry, the application window has closed.
      </span>
    </div>
  </div>
</div>
`;

exports[`dashboard incomplete before deadline 1`] = `
<div
  className="dashboard"
  style={
    Object {
      "backgroundImage": "url('function SvgCombined_circuit(props) {
  return React.createElement(
    'svg', 
    Object.assign({}, props, {'data-file-name': SvgCombined_circuit.name})
  );
}')",
    }
  }
>
  <div
    style={
      Object {
        "left": "50%",
        "position": "absolute",
        "top": "50%",
        "transform": "translateX(-50%) translateY(-50%)",
      }
    }
  >
    <div
      className="dashboard-design"
    >
      <div>
        <div>
          You haven't submitted your application yet. You have
        </div>
        <div
          style={
            Object {
              "color": "#00E073",
              "fontSize": "70px",
              "marginBottom": -15,
            }
          }
        >
          7263
        </div>
        <div>
          days to submit your application before the deadline:
          <br />
          <strong>
            November 19, 2018
          </strong>
          .
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`dashboard submitted 1`] = `
<div
  class="dashboard"
  style="background-image:url('function SvgCombined_circuit(props) {
  return React.createElement(
    'svg', 
    Object.assign({}, props, {'data-file-name': SvgCombined_circuit.name})
  );
}')"
>
  <div
    style="position:absolute;top:50%;left:50%;transform:translateX(-50%) translateY(-50%)"
  >
    <div
      class="dashboard-design"
    >
      <span>
        Your application has been received â€“ you are all good for now!
        <br />
        <br />
        We will email you when decisions are released and will handle any travel questions at that time. Thanks for applying :)
      </span>
    </div>
  </div>
</div>
`;
